src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -139,7 +139,14 @@ public function setup_meta( $meta ) {
 	 */
 	public function update_meta( $key, $value ) {
 		$this->meta[ $key ] = $value;
-		return update_post_meta( $this->post->ID, self::$meta_key_prefix . $key, $value );
+
+		$field = self::$meta_key_prefix . $key;
+
+		if ( null === $value ) {
```

Comment:
You've got a better idea of what's happening in here than I do - but could it be worth testing if the value is empty rather than strictly null?